pkg/kgo/config.go

```diff
@@ -388,9 +388,36 @@ func (cfg *cfg) validate() error {
 		return errors.New("invalid group partition assigned/revoked/lost functions set when a group was not specified")
 	}
 
+	processedHooks, err := processHooks(cfg.hooks)
+	if err != nil {
+		return err
+	}
+	cfg.hooks = processedHooks
+
 	return nil
 }
 
+// processHooks will inspect and recusively unpack slices of hooks stopping
+// if the instance implements any hook interface. It will return an error on
+// the first instance that implements no hook interface
+func processHooks(hooks []Hook) ([]Hook, error) {
+	processedHooks := []Hook{}
```

Comment:
nit: I generally prefer `var processedHooks []Hook` (only adding this nit because of the other comment)